### ISSUE TITLE
[Feature Request]: Kopia should follow a .kopiaignore symlink.

### ISSUE THREAD

**Issue by dsimmons** [kopia/kopia#2037]
Creating an issue to capture [this discussion](https://kopia.discourse.group/t/kopia-ignores-kopiaignore-symlink/1210) on the Kopia Discourse forum with @jkowalski. Per his reply:

> "It’s not intentional, but Kopia indeed expects this to be a file and not symlink today. Happy to review PR to fix it."

What follows is an embedding of the original post I wrote up for convenience:

---

### Behavior

Suppose I have a `.kopiaignore` file located in the root of my home directory, `~/.kopiaignore`.

In the service of keeping the exemplified use case simple, let's have the contents read:

```bash
.cache
```

If I then do:

```bash
kopia snapshot estimate $HOME
```

... we verify that the `~/.cache` dir was indeed excluded! In other words, this works as intended.

Now, let's suppose instead I have my "canonical" `.kopiaignore` located at `~/dotfiles/kopia/.kopiaignore` and I symlink that to my home directory, such that it looks like:

```bash
# ls -la ~
lrwxrwxrwx 1 user user 27 June 12 12:17 .kopiaignore -> dotfiles/kopia/.kopiaignore
```

If I again do:

```bash
kopia snapshot estimate $HOME
```

... we observe that none of the exclusions/ignores take effect!

### Commentary

While obviously the above is a bit of a contrived (simple) example, it definitely caught me off-guard in terms of a real world use case just now:

![2022-06-12-122216_1745x91_scrot|690x35](https://canada1.discourse-cdn.com/free1/uploads/kopia/original/1X/4b93a24b88c7e3a0620e935eb3ed88cd5c2fa4b5.png)

Notice how, with the top snapshot, the number of files is roughly ~178K. Then, after making the above change, swapping a static file to a symlink, all of my exclusions were ignored for the subsequent snapshot (and we see a jump up of almost ~600K files and 17GB).

### Use Case

I have a `dotfiles` repository/dir (in the Git sense) that lives within my home directory, `~/dotfiles`. This directory is synced between machines that I use and/or work on regularly.

I use [GNU Stow](https://www.gnu.org/software/stow/) to "install" (symlink) these dotfiles onto machines.

Example:

```bash
pwd
# ~/dotfiles

ls -l
# alacritty
# git
# gnupg
# ...
# kopia
# ...

stow kopia
cd

ls -l
# .kopiaignore -> dotfiles/kopia/.kopiaignore
```

**TL;DR:** This symlinking approach allows me to keep my Kopia exclusions in-sync between machines without having to update/overwrite `N` different static `.kopiaignore` files.

For what it's worth, I use this pattern to great success with nearly all of my "daily driver" binaries!

**Comment by ldtjcdrs** [comment]
Another use for this feature is to allow the `.kopiaignore` file to be managed declaratively by Nix / NixOS.

**Comment by mcamou** [comment]
I would like to tackle this. Could you please assign me and add the `hacktoberfest-accepted` label?

**Comment by dsimmons** [comment]
cc @julio-lopez, doesn't look like I'm able to make the requested changes to the issue on my end!

Greatly appreciated @mcamou :slightly_smiling_face:

**Comment by dsimmons** [comment]
Addressed as part of #4190, sincerely appreciate you tackling this @mcamou! :saluting_face: 

Just installed the latest Kopia version locally and verified it worked as expected, big quality of life increase! And in fact, my non-symlinked `.kopiaignore` had drifted from the version checked into SCM as I'd made some one-off exclusions (case and point!).

Going ahead and closing this issue accordingly.
